tests/serialize/runstate/dynamodb_state_store_test.py

```diff
@@ -1,108 +1,50 @@
-            name,
-            key,
-            update_expression,
-            attribute_updates,
-            expression_attribute_names,
-            expression_attribute_values,
-        )
+@pytest.fixture
+def table_name():
+    return "test-tron-state"
+
 
-    transact_items = self.body["TransactItems"]
-
-    for transact_item in transact_items:
-        if "Put" in transact_item:
-            put_item(transact_item["Put"])
-        elif "Update" in transact_item:
-            update_item(transact_item["Update"])
-        elif "Delete" in transact_item:
-            delete_item(transact_item["Delete"])
-
-    return dynamo_json_dump({})
-
-
-@pytest.fixture(autouse=True)
-def store():
-    with mock.patch(
-        "moto.dynamodb2.responses.DynamoHandler.transact_write_items",
-        new=mock_transact_write_items,
-        create=True,
-    ), mock_dynamodb2():
-        dynamodb = boto3.resource("dynamodb", region_name="us-west-2")
-        table_name = "tmp"
-        store = DynamoDBStateStore(table_name, "us-west-2", stopping=True)
-        store.table = dynamodb.create_table(
+@pytest.fixture
+def store(table_name):
+    """Provides a DynamoDBStateStore instance with a mocked DynamoDB backend."""
+    with mock_dynamodb2():
+        client = boto3.client("dynamodb", region_name="us-west-2")
+
+        client.create_table(
             TableName=table_name,
             KeySchema=[
-                {
-                    "AttributeName": "key",
-                    "KeyType": "HASH",
-                },  # Partition key
-                {
-                    "AttributeName": "index",
-                    "KeyType": "RANGE",
-                },  # Sort key
+                {"AttributeName": "key", "KeyType": "HASH"},
+                {"AttributeName": "index", "KeyType": "RANGE"},
             ],
             AttributeDefinitions=[
-                {
-                    "AttributeName": "key",
-                    "AttributeType": "S",
-                },
-                {
-                    "AttributeName": "index",
-                    "AttributeType": "N",
-                },
+                {"AttributeName": "key", "AttributeType": "S"},
+                {"AttributeName": "index", "AttributeType": "N"},
             ],
-            ProvisionedThroughput={
-                "ReadCapacityUnits": 10,
-                "WriteCapacityUnits": 10,
-            },
+            ProvisionedThroughput={"ReadCapacityUnits": 10, "WriteCapacityUnits": 10},
         )
-        store.client = boto3.client("dynamodb", region_name="us-west-2")
-        # Has to be yield here for moto to work
-        yield store
+
+        # Pass `stopping=True` to prevent the background save thread from running during tests
+        state_store = DynamoDBStateStore(table_name, "us-west-2", stopping=True)
+        yield state_store
 
 
 @pytest.fixture
-def small_object():
-    yield {
+def job_state_object():
+    return {"enabled": True, "run_nums": [0, 1]}
```

Comment:
it's nice to always yield in fixtures since that's one less change to make if we ever need some post-fixture cleanup - but this is also fine as-is :)